## Lab Assignment One: Exploring Table Data

#### Luis Garduno

### 1. Business Understanding

##### About League of Legends :
Developed by Riot Studios, League of Legends, or "LoL", is an online multiplayer
video-game that is available to Windows/MacOS users. LoL consists 2 teams ('Blue
& 'Red') facing each other, where the main objective is to destroy the opposing
teams 'Nexus', or home base, while facing obstacles like destroying towers &
eliminating players throughout the way. Perks & gold are able to be obtained by
players/teams through completing tasks such as eliminating players, enemy creeps,
or dragons. Players then spend the gold to purchase items that help deal more
damage towards enemies.

League of Legends offers different game modes, such as ranked. A ranked game on
average lasts 30-45 minutes. In this game mode, players are given a rank based
off of the number of wins + the number of games played. "Diamond" is one of the
highest ranks a player may obtain and is known to be extremely competitive. The
dataset contains the first 10 minute analytics of each team for different
diamond ranked matches.

Once the data is analyzed, teams/players would be able to conceptualize the
level of priority different features/objectives have during early stages of
diamond ranked matches. They could then use this information to adjust their
strategy to one proven to win matches.

-------------------------------------

Dataset : https://www.kaggle.com/bobbyscience/league-of-legends-diamond-ranked-games-10-min

Question Of Interest : As of the first 10 minutes, which team will win?

### 2. Data Understanding

In [1]:
import numpy as np
import pandas as pd

# Load in the dataset into a dataframe
df = pd.read_csv('https://raw.githubusercontent.com/luisegarduno/MachineLearning_Projects/master/Datasets/high_diamond_ranked_10min.csv')

# delete gameID column
del df['gameId']

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9879 entries, 0 to 9878
Data columns (total 39 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   blueWins                      9879 non-null   int64  
 1   blueWardsPlaced               9879 non-null   int64  
 2   blueWardsDestroyed            9879 non-null   int64  
 3   blueFirstBlood                9879 non-null   int64  
 4   blueKills                     9879 non-null   int64  
 5   blueDeaths                    9879 non-null   int64  
 6   blueAssists                   9879 non-null   int64  
 7   blueEliteMonsters             9879 non-null   int64  
 8   blueDragons                   9879 non-null   int64  
 9   blueHeralds                   9879 non-null   int64  
 10  blueTowersDestroyed           9879 non-null   int64  
 11  blueTotalGold                 9879 non-null   int64  
 12  blueAvgLevel                  9879 non-null   float64
 13  blu

Printing out the information about the dataframe we able to see that there are a
total of 9,879 instances, and 39 attributes. The "gameId" column is removed
since it will have no impact on the outcome.

Additionally we are able to see that there are 19 of the same attributes for each
the blue & red team (columns 1-19 are the same as 20-38). Features for each team includes :
- Wards placed & destroyed
- Total number of kills, deaths, & assists
- First Bloods (1st elimination of the game)
- Total : towers destroyed, gold, experience
- Average : level, CS per minute, & gold per minute
- Difference in gold & experience between the teams
- Objective kills : elite monsters, dragons, heralds, minions, & jungle minions

Lastly, when looking at the data types and total number of entries, we see that there
are no missing entries and that all of the data types are already correct.

A majority of the columns will be of type integer (int64) because they will always be
whole number. Attributes involving averages such as cs per minute, gold per minute, &
level will be double-precision floating-point format (float64).

The winner of the match is also included as one of the attributes, listed as
"blueWins" : where '1' means true (blue team won) & '0' meaning false (red team won).
Also, "first bloods" only occur once per game so that will also be either true or false.
But because we are wanting to visualize these attributes, optimally it is best to
keep these as integer data types.

----------------------

// [1.5 points] Verify data quality:
- Explain any missing values or duplicate data.
- Visualize entries that are missing/complete for different attributes.
- Are those mistakes?
- Why do these quality issues exist in the data?
- How do you deal with these problems?
- Give justifications for your methods (elimination or imputation).


### 3. Data Visualization
// [2 points] Visualize basic feature distributions.
- That is, plot the dynamic range and exploratory distribution plots (like boxplots, histograms, kernel density estimation) to better understand the data.
- Describe anything meaningful or potentially useful you discover from these visualizations.
- These may also help to understand what data is missing or needs imputation.
- Note: You can also use data from other sources to bolster visualizations.
- Visualize at least five plots, at least one categorical and at least one numeric.

// [2.5 points] (3.1 Data Exploration) (3.2 Data Relationship Exploration)
- Ask 3 interesting questions that are relevant to your dataset & explore visuals that help answer these questions.
- Use whichever visualization method is appropriate for your data.
- Important: Interpret the implications for each visualization.

### 4. Dimensionality Reduction
#### Reference